spag.py

```diff
@@ -7,37 +7,49 @@
 
 import click
 import requests
+import yaml
 
 class ToughNoodles(Exception):
     pass
 
-class Endpoint(object):
+class Environment(object):
 
-    ENDPOINT_FILE = '.spag.endpoint'
+    ENV_FILE = '.env.yml'
     @classmethod
     def get(cls):
-        if os.path.exists(cls.ENDPOINT_FILE):
-            with click.open_file(cls.ENDPOINT_FILE, 'r') as f:
-                return f.read()
+        if os.path.exists(cls.ENV_FILE):
+            with click.open_file(cls.ENV_FILE, 'r') as f:
+                return yaml.load(f)
         else:
             raise ToughNoodles("Endpoint not set")
 
     @classmethod
-    def set(cls, endpoint):
-        with click.open_file(cls.ENDPOINT_FILE, 'w') as f:
-            f.write(endpoint)
+    def set(cls, **kwargs):
+        # This ignores any arguments that are set to None
+        kwargs = {key: value for key, value in kwargs.items() if value is not None}
+
+        if os.path.exists(cls.ENV_FILE):
+            f = click.open_file(cls.ENV_FILE, 'r')
+            data = yaml.load(f)
+            data.update(kwargs)
+        else:
+            data = kwargs
+
+        f = click.open_file(cls.ENV_FILE, 'w+')
+        yaml.safe_dump(data, f)
```

Comment:
Apparently, if you `yaml.dump(x, default_flow_style=False)` it will get rid of the braces.